src/Propel/Generator/Model/ForeignKey.php

```diff
@@ -625,9 +640,8 @@ public function getColumnObjectsMapping()
      */
     public function getMappedForeignColumn($local)
     {
-        $m = $this->getLocalForeignMapping();
-
-        return isset($m[$local]) ? $m[$local] : null;
+        $index = array_search($local, $this->localColumns);
+        return $this->foreignColumns[$index];
```

Comment:
@marcj  

should this return null when `$local` could not be found using `array_search` instead of issue a php warnig because of a undefined index `$this->foreignColumns[$index]`?

same below in `getMappedLocalColumn()`